ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/forkchoice/ForkChoiceBlobSidecarsAvailabilityChecker.java

```diff
@@ -109,12 +149,212 @@ private BlobSidecarsAndValidationResult internalValidate(final List<BlobSidecar>
+
+    final List<KZGCommitment> kzgCommitmentsInBlock = kzgCommitmentsSupplier.get();
+    final SortedMap<UInt64, BlobSidecar> completeBlobSidecars =
+        blockBlobSidecarsTracker.getBlobSidecars();
+
+    UInt64.range(UInt64.ZERO, UInt64.valueOf(kzgCommitmentsInBlock.size()))
+        .forEachOrdered(
+            index -> {
+              if (!validatedBlobSidecars.containsKey(index)) {
+                additionalKzgCommitmentsToBeValidated.add(
+                    kzgCommitmentsInBlock.get(index.intValue()));
+
+                Optional.ofNullable(completeBlobSidecars.get(index))
+                    .ifPresentOrElse(
+                        additionalBlobSidecarsToBeValidated::add,
+                        () ->
+                            LOG.error(
+                                "Index {} not found in blob sidecar returned by tracker for block {}",
+                                index,
+                                blockBlobSidecarsTracker.getSlotAndBlockRoot().toLogString()));
+              }
+            });
+
+    return internalValidate(
+        additionalBlobSidecarsToBeValidated, additionalKzgCommitmentsToBeValidated);
+  }
+
+  /**
+   * Computes the final validation result combining the already validated blobs from
+   * `validatedBlobSidecars` and the additional validation result containing (if validated) the
+   * required additional blobs to reconstruct the full list that will have to match the kzg
+   * commitments from the block
+   *
+   * @param additionalBlobSidecarsAndValidationResult
+   * @return
+   */
+  private BlobSidecarsAndValidationResult internalComputeFinalValidationResult(
+      final BlobSidecarsAndValidationResult additionalBlobSidecarsAndValidationResult) {
+    if (additionalBlobSidecarsAndValidationResult.isFailure()) {
+      return additionalBlobSidecarsAndValidationResult;
+    }
+
+    // put all additional validated blobSidecar in validatedBlobSidecars which now should
+    // be complete
+    additionalBlobSidecarsAndValidationResult
+        .getBlobSidecars()
+        .forEach(blobSidecar -> validatedBlobSidecars.put(blobSidecar.getIndex(), blobSidecar));
+
+    // let's create the final list of validated BlobSidecars making sure that indices and
+    // final order are consistent
+    final List<BlobSidecar> completeValidatedBlobSidecars = new ArrayList<>();
+    validatedBlobSidecars.forEach(
+        (index, blobSidecar) -> {
+          checkState(
+              index.equals(blobSidecar.getIndex())
+                  && index.equals(UInt64.valueOf(completeValidatedBlobSidecars.size())),
+              "Inconsistency detected during blob sidecar validation");
+          completeValidatedBlobSidecars.add(blobSidecar);
+        });
+
+    if (completeValidatedBlobSidecars.size() < kzgCommitmentsSupplier.get().size()) {
+      // we haven't verified enough blobs to match the commitments present in the block
+
+      checkState(
```

Comment:
I'd add some comment, why we could have such state here to pass it to NOT_REQUIRED